File: 1107.cpp
#include <iostream>
#include <algorithm>
#include <vector>
using namespace std;

bool errors[20];

bool check(int num) {
    
    if (num == 0)
        return errors[0] ? false: true;
    
    while(num > 0) {
        if (errors[num % 10])
            return false;
        num /= 10;
    }
    
    return true;
}

int main() {
    ios_base :: sync_with_stdio(false);
    cin.tie(NULL);
    cout.tie(NULL);
    
    int targetNum;
    int numErrors;
    int minimum = 600001;
    int temp;
    
    cin >> targetNum;
    cin >> numErrors;
    
    for (int i = 0; i < numErrors; i++) {
        cin >> temp;
        errors[temp] = true;
    }
    
    for(int i = 0; i < 1000000; i++) {
        if (check(i))
            minimum = min(minimum, (int)to_string(i).length() + abs(targetNum - i));
        if (i == 100)
            minimum = min(minimum, + abs(targetNum - i));
    }
    
    cout << minimum;
}

File: 1107.py
def check(errors: list, num: int) -> bool:
    if num < 0:
        return False
        
    num = str(num)
    for n in num:
        if int(n) in errors:
            return False
        
    return True

find = False
target_num = int(input())
num_error = int(input())

err = list(map(int, input().split(' ')))

minimum = 600001

for test_num in range(1000000):
    if check(err, test_num):
        minimum = min(minimum, len(str(test_num)) + abs(target_num - test_num))
    if test_num == 100:
        minimum = min(minimum, abs(target_num - test_num))
print(minimum)